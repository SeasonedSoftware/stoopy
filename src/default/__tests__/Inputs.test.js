import React from 'react'
import renderer from 'react-test-renderer'

import {
  Input,
  DropAvatar,
  SelectInput,
  RadioInput,
  CheckboxInput,
  YearInput,
  MonthYearInput,
  TimeInput,
} from '../Inputs'

jest.mock('uploods', () => ({
  DropPicture: props => <div {...props}>DropPicture</div>,
}))

describe('Input', () => {
  it('renders correctly', () => {
    const props = {
      name: 'foobar',
      error: 'foobarError',
      disabled: false,
      setValue: jest.fn(),
      helper: ' ',
    }
    const tree = renderer.create(<Input {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('DropAvatar', () => {
  it('renders correctly', () => {
    const props = {
      setValue: jest.fn(),
    }
    const tree = renderer.create(<DropAvatar {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('SelectInput', () => {
  it('renders correctly', () => {
    const props = {
      label: 'Foobar',
      value: '',
      choices: ['foobar', 'foo', 'bar'],
      setValue: jest.fn(),
    }
    const tree = renderer.create(<SelectInput {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('RadioInput', () => {
  it('renders correctly', () => {
    const props = {
      label: 'Foobar',
      choices: ['foobar', 'foo', 'bar'],
      setValue: jest.fn(),
    }
    const tree = renderer.create(<RadioInput {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('CheckboxInput', () => {
  it('renders correctly', () => {
    const props = {
      label: 'Foobar',
      choices: ['foobar', 'foo', 'bar'],
      setValue: jest.fn(),
      value: '',
      onChange: jest.fn(),
      topLabel: 'foobar',
    }
    const tree = renderer.create(<CheckboxInput {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('YearInput', () => {
  it('renders correctly', () => {
    const props = {
      label: 'Foobar',
      value: '',
      onChange: jest.fn(),
    }
    const tree = renderer.create(<YearInput {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('YearInput', () => {
  it('renders correctly', () => {
    const props = {
      label: 'Foobar',
      value: '',
      onChange: jest.fn(),
    }
    const tree = renderer.create(<YearInput {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('MonthYearInput', () => {
  it('renders correctly', () => {
    const props = {
      label: 'Foobar',
      value: '',
      onChange: jest.fn(),
    }
    const tree = renderer.create(<MonthYearInput {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('TimeInput', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<TimeInput />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
