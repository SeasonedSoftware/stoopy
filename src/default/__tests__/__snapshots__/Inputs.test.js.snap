// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxInput renders correctly 1`] = `
<fieldset
  className="MuiFormControl-root"
>
  <legend
    className="MuiFormLabel-root"
  >
    foobar
  </legend>
  <div
    className="MuiFormGroup-root"
  >
    <label
      className="MuiFormControlLabel-root"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-104 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <input
            checked={false}
            className="PrivateSwitchBase-input-107"
            data-indeterminate={false}
            disabled={false}
            onChange={[Function]}
            type="checkbox"
            value="true"
          />
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </span>
      <span
        className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
      >
        Foobar
      </span>
    </label>
  </div>
</fieldset>
`;

exports[`Input renders correctly 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
  style={
    Object {
      "marginBottom": 10,
    }
  }
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-error Mui-error"
    data-shrink={false}
    htmlFor="foobar"
    id="foobar-label"
  >
    Foobar
  </label>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline Mui-error Mui-error MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <input
      aria-describedby="foobar-helper-text"
      aria-invalid={true}
      autoFocus={false}
      className="MuiInputBase-input MuiInput-input"
      disabled={false}
      id="foobar"
      name="foobar"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Foobar"
      required={false}
      type="text"
    />
  </div>
  <p
    className="MuiFormHelperText-root Mui-error"
    id="foobar-helper-text"
  >
    foobarError
  </p>
</div>
`;

exports[`RadioInput renders correctly 1`] = `
<fieldset
  className="MuiFormControl-root"
>
  <legend
    className="MuiFormLabel-root"
  >
    Foobar
  </legend>
  <div
    className="MuiFormGroup-root"
    role="radiogroup"
  >
    <label
      className="MuiFormControlLabel-root"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-104 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <input
            checked={false}
            className="PrivateSwitchBase-input-107"
            disabled={false}
            onChange={[Function]}
            type="radio"
            value="foobar"
          />
          <div
            className="PrivateRadioButtonIcon-root-120"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root PrivateRadioButtonIcon-layer-121"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
              />
            </svg>
          </div>
        </span>
      </span>
      <span
        className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
      >
        foobar
      </span>
    </label>
    <label
      className="MuiFormControlLabel-root"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-104 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <input
            checked={false}
            className="PrivateSwitchBase-input-107"
            disabled={false}
            onChange={[Function]}
            type="radio"
            value="foo"
          />
          <div
            className="PrivateRadioButtonIcon-root-120"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root PrivateRadioButtonIcon-layer-121"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
              />
            </svg>
          </div>
        </span>
      </span>
      <span
        className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
      >
        foo
      </span>
    </label>
    <label
      className="MuiFormControlLabel-root"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-104 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <input
            checked={false}
            className="PrivateSwitchBase-input-107"
            disabled={false}
            onChange={[Function]}
            type="radio"
            value="bar"
          />
          <div
            className="PrivateRadioButtonIcon-root-120"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root PrivateRadioButtonIcon-layer-121"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
              />
            </svg>
          </div>
        </span>
      </span>
      <span
        className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
      >
        bar
      </span>
    </label>
  </div>
</fieldset>
`;

exports[`SelectInput renders correctly 1`] = `
<div
  className="MuiFormControl-root"
  style={
    Object {
      "minWidth": 120,
    }
  }
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
    data-shrink={false}
  >
    Foobar
  </label>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <div
      aria-haspopup="listbox"
      aria-labelledby=" "
      className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "&#8203;",
          }
        }
      />
    </div>
    <input
      onAnimationStart={[Function]}
      type="hidden"
      value=""
    />
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root MuiSelect-icon"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;
